chain/sync.go

```diff
@@ -723,10 +725,27 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		return err
 	}
 
+	defer func() {
+		ntfnCtxCancel()
+
+		select {
+		case <-ctx.Done():
+			wsClient.Close()
+		default:
+		}
+
+		// Wait for notifications to finish before returning
+		<-s.notifier.closed
+	}()
```

Comment:
Aren't we already canceling `ntfnCtx` with above `defer` (why do it here as well) ?

Same for `wsClient.Close()` (this `select` doesn't seem to be doing anything)